pkgs/by-name/si/sile/package.nix

```diff
@@ -0,0 +1,181 @@
+{
+  lib,
+  stdenv,
+  fetchzip,
+
+  # nativeBuildInputs
+  pkg-config,
+  jq,
+  cargo,
+  rustc,
+  rustPlatform,
+
+  # buildInputs
+  lua,
+  harfbuzz,
+  icu,
+  fontconfig,
+  libiconv,
+  stylua,
+  typos,
+  darwin,
+  # FONTCONFIG_FILE
+  makeFontsConf,
+  gentium,
+
+  # passthru.tests
+  runCommand,
+  poppler_utils,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "sile";
+  version = "0.15.5";
+
+  src = fetchzip {
+    url = "https://github.com/sile-typesetter/sile/releases/download/v${finalAttrs.version}/sile-${finalAttrs.version}.zip";
+    sha256 = "sha256-zP+MGCXGEg19U6tMrHIdgAAfKQT21vFtmoEROXgxUB0=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) src;
+    dontConfigure = true;
+    hash = "sha256-hmgDG29C5JfQX2acMr8c3lmswa1u5XHauRWFd4QGmOo=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    jq
+    cargo
+    rustc
+    rustPlatform.cargoSetupHook
+  ];
+
+  buildInputs =
+    [
+      finalAttrs.finalPackage.passthru.luaEnv
+      harfbuzz
+      icu
+      fontconfig
+      libiconv
+      stylua
+      typos
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      darwin.apple_sdk.frameworks.AppKit
+    ];
+
+  configureFlags =
+    [
+      # Nix will supply all the Lua dependencies, so stop the build system from
+      # bundling vendored copies of them.
+      "--with-system-lua-sources"
+      "--with-system-luarocks"
+      # The automake check target uses pdfinfo to confirm the output of a test
+      # run, and uses autotools to discover it. This flake build eschews that
+      # test because it is run from the source directory but the binary is
+      # already built with system paths, so it can't be checked under Nix until
+      # after install. After install the Makefile isn't available of course, so
+      # we have our own copy of it with a hard coded path to `pdfinfo`. By
+      # specifying some binary here we skip the configure time test for
+      # `pdfinfo`, by using `false` we make sure that if it is expected during
+      # build time we would fail to build since we only provide it at test time.
+      "PDFINFO=false"
+    ]
+    ++ lib.optionals (!lua.pkgs.isLuaJIT) [
+      "--without-luajit"
+    ];
+
+  outputs = [
+    "out"
+    "doc"
+    "man"
+    "dev"
+  ];
```

Comment:
In many packages, `configureFlags` do something related to the `outputs`, so it is natural to put these right after `configureFlags`.